tests/test_cases/coreutils_suite/shm_sem_client_fork.c

```diff
@@ -0,0 +1,78 @@
+#include <sys/ipc.h>
+#include <sys/shm.h>
+#include <stdio.h>
+#include <string.h>
+#include <semaphore.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+int main()
+{
+    key_t key = 31337;
+    // shmget returns an identifier in shmid
+    int shmid = shmget(key, 1024, 0666);
+
+    // shmat to attach to shared memory
+    sem_t *sem_ptr = (sem_t *)shmat(shmid, NULL, 0);
+
+    key_t key2 = 13371;
+    // shmget returns an identifier in shmid
+    int shmid2 = shmget(key2, 1024, 0666);
+
+    // shmat to attach to shared memory
+    char *shared_buffer = (char *)shmat(shmid2, NULL, 0);
+
+    if (fork() == 0) {
+        // child process
+        sleep(0.5);
+        int count = 0;
+        while (count++ < 3)
+        {
+            int status = sem_wait(sem_ptr);
+
+            if (status != 0)
+            {
+                perror("in sem_wait");
```

Comment:
This modification adds tests for the use of semaphores in shared memory, including multiple processes. All look fantastic! But I'm a bit confused: should the test end when an error occurs?

Reply:
Ideally yeah it should end right? Do you that instead of throwing an error just end it using an `exit(-1)`

Reply:
Oh im just wondering whether program ends after perror

Reply:
I added an `exit(1)` after all the perrors. As you said it might just be better to exit if there is an error.